Check for unsupported target options even with -Qunused-arguments

MatzeB opened the pull request:
Fix clang accidentally skipping checks for `err_drv_unsupported_opt_for_target` when `-Qunused-arguments` was active.

MatzeB committed:
Check for unsupported target options even with -Qunused-arguments

MaskRay commented on clang/test/Driver/unsupported-option.c:
```diff
@@ -27,3 +27,10 @@
 // UNSUP_OPT: error: unsupported option
+
+
+// RUN: not %clang -c --target=aarch64-- -mfpu=crypto-neon-fp-armv8 %s 2>&1 | \
```
This RUN line without -Qunused-arguments doesn't add test coverage. It can be removed

MaskRay commented on clang/test/Driver/unsupported-option.c:
```diff
@@ -27,3 +27,10 @@
+// RUN: not %clang -c --target=aarch64-- -mfpu=crypto-neon-fp-armv8 %s 2>&1 | \
+// RUN: FileCheck %s --check-prefix=AARCH64
+// RUN: not %clang -c -Qunused-arguments --target=aarch64-- -mfpu=crypto-neon-fp-armv8 %s 2>&1 | \
+// RUN: FileCheck %s --check-prefix=AARCH64
```
indent the continuation line by 2 spaces

MatzeB committed:
address review feedback, change logic to check for HadError earlier

- Address review feedback
- Change logic to check for `HasError()` before the loop again to be
  closer to the original behavior; fixes `x86-target-features.c`.
